tests/test_znframe.py

```diff
@@ -55,3 +66,8 @@ def test_frame_from_dict(ammonia):
 
 def test_to_json(ammonia):
     assert Frame.from_json(ammonia.to_json()) == ammonia
+
+
+def test_water_with_calc(waterWithCalc):
+    frame = Frame.from_atoms(waterWithCalc)
+    assert frame.to_atoms().calc == waterWithCalc.calc
```

Comment:
Maybe check, that there are no duplicates one the `frame` is created in `arrays` and `calc` here as well. Then all the new features should be covered.